course/2 Neural Network Architectures/2.1 Convolutional Neural Networks/2.1.2 Common Pre-Trained Models/vgg.md

```diff
@@ -0,0 +1,62 @@
+
+# Very Deep Convolutional Networks for Large Scale Image Recognition (2014)
+
+
+1. Introduction
+
+VGG architecture was developed in 2014 by Karen Simonyan and Andrew Zisserman from Visual Geometry Group at Oxford University. The authors detailed key advancements in convolutional neural networks for ImageNet Large Scale Visual Recognition Challenge (ILSVRC). VGG demonstrated significant improvements with increased depth and repeated small (3X3) convolutional filters. It was a research project to systematically evaluate prior-art configurations and identify key structural designs. 
+
+
+
+2. VGG Network Architechture 
+   
+- Input is 224X224 image. 
+- 3X3 convolution kernels with 2X2 max pooling layers.
+- Number of channels for each convolutional layer 64 -> 128 -> 256 -> 512 -> 512. 
+- VGG16 has 16 hidden layers (13 convolutional layers and 3 fully connected layers).
+- VGG19 has 19 hidden layers (16 convolutional layers and 3 fully connected layers).
+
+Below is network structure of VGG16, The activation and dropout layers were omitted for brevity. MaxPooling layers were included to display change in output tensor size at each layer. 
+
+|Layer Type |Filter Size|Output Size|No. Params|
+|-----------|------------|----------|----------|
+|Input Image|0|3X224X224|0|
+|Conv2D|3X3|64X224X224|1,792|
+|Conv2D|3X3|64X224X224|39,628|
+|MaxPool2D|2X2|64X112X112|0|
+|Conv2D|3X3|128X112X112|73,856|
+|Conv2D|3X3|128X112X112|147,584|
+|MaxPool2D|2X2|128X56X56|0|
+|Conv2D|3X3|256X56X56|295,168|
+|Conv2D|3X3|256X56X56|590,080|
+|Conv2D|3X3|256X56X56|590,080|
+|MaxPool2D|2X2|256X28X28|0|
+|Conv2D|3X3|512X28X28|1,180,160|
+|Conv2D|3X3|512X28X28|2,359,808|
+|Conv2D|3X3|512X28X28|2,359,808|
+|MaxPool2D|2X2|512X14X14|0|
+|Conv2D|3X3|512X14X14|2,359,808|
+|Conv2D|3X3|512X14X14|2,359,808|
+|Conv2D|3X3|512X14X14|2,359,808|
+|MaxPool2D|2X2|0|
+|Linear|N/A|4096|102,764,544|
+|Linear|N/A|4096|16,781,312|
+|Linear|N/A|1000|4,096,000|
```

Comment:
I find making these like block diagram would be better. no need to change in this PR. we can make another PR after merging this one, for adding a diagram. what do you think @albertkao227 ?

Reply:
@sezan92  Yes I believe a diagram is better. We should make it or find a good one and reference.